src/components/implementation/capmgr/naive/mem_mgr.c

```diff
@@ -65,3 +65,35 @@ memmgr_shared_page_map_cserialized(vaddr_t *pgaddr, int *unused, int idx)
 done:
 	return num_pages;
 }
+
+/* These functions return the address at which the gs register is set to */
+void *
+memmgr_tls_alloc(unsigned int dst_tid)
+{
+	/* Just return the vaddr range for this tid */
+	assert(dst_tid < MAX_NUM_THREADS);
+	return TLS_BASE_ADDR + (void *)(TLS_AREA_SIZE * dst_tid);
+}
+
+void *
+_memmgr_tls_alloc_and_set(void *area)
```

Comment:
Why is the weird naming `_memmgr_xxxx`? 

Passing address `area` between address-spaces? If it's not used, why pass it?

Reply:
It should not be passed in here

Reply:
I don't see a benefit of having 2 interfaces one making the caller (client) do cos_thd_mod and the other doing it from the `capmgr`. 

The point of the first interface I thought originally was really just get the address for being able to access it in your component. (It's my bad I didn't follow the PR description thoroughly).

We talked about it briefly in person but I still think we should just have a single interface. 
I've a suggestion above, which is to make `cos_thd_mod` be part of the thread creation and this interface is only for getting the address! That would be great. If we were to have a tls manager as a separate component, then this detail of actually cos_thd_mod() could still be hidden in that one interface for tls allocation or set! But I'm not sure if we'd want a standalone tls manager either..